Assets/Scripts/Abilities/AbstractAbility.cs

```diff
@@ -0,0 +1,37 @@
+using System;
+using UnityEngine;
+using UnityEngine.InputSystem;
+
+public abstract class AbstractAbility
+{
+    protected string _animatorID;
+    protected Rigidbody2D _body;
+    protected Animator _animator;
+    protected bool _active;
+    protected float _energyCost;
+
+    public static event Action<float> AbilityTriggered;
```

Comment:
I think the children might end up sharing this on accident based on https://stackoverflow.com/questions/3542171/c-sharp-abstract-class-static-field-inheritance